components/engine_rocks/src/properties.rs

```diff
@@ -347,14 +393,28 @@ impl RangePropertiesCollector {
 
 impl TablePropertiesCollector for RangePropertiesCollector {
     fn add(&mut self, key: &[u8], value: &[u8], entry_type: DBEntryType, _: u64, _: u64) {
-        // size
-        let size = match get_entry_size(value, entry_type) {
-            Ok(entry_size) => key.len() as u64 + entry_size,
-            Err(_) => return,
-        };
-        self.cur_offsets.size += size;
-        // keys
-        self.cur_offsets.keys += 1;
+        if entry_type != DBEntryType::Delete {
+            let size = match get_entry_size(value, entry_type) {
+                Ok(entry_size) => key.len() as u64 + entry_size,
+                Err(_) => return,
+            };
+            self.cur_offsets.size += size;
+            self.cur_offsets.keys += 1;
+        } else {
+            // handle delete entry type
+            self.cur_offsets.delete_keys += 1;
+            // Consider that we firstly insert a new point with a key, then delete the same key,
+            // the information of deleted keys should be changed in the last offset.
+            if let Some(last_point) = self.props.offsets.last() {
+                if last_point.0 == key {
+                    let mut last_offset = last_point.1.clone();
+                    last_offset.delete_keys = self.cur_offsets.delete_keys;
+                    self.props.offsets.pop();
+                    self.props.offsets.push((key.to_owned(), last_offset));
+                }
+            }
+        }
+
         // Add the start key for convenience.
         if self.last_key.is_empty()
             || self.size_in_last_range() >= self.prop_size_index_distance
```

Comment:
delete should trigger a new point too, consider the case in which an SST has only delete entries.